MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -7146,18 +7146,38 @@ public void assignRandomPortraitFor(Person p) {
         }
         ArrayList<String> possiblePortraits = new ArrayList<String>();
         Iterator<String> categories = portraits.getCategoryNames();
-        
+
+        // Will search for portraits in the /gender/primaryrole folder first,
+        // and if none are found then /gender/rolegroup, then /gender/combat or
+        // /gender/support, then in /gender.
         String searchCat_Gender = "";
         if (p.getGender() == Person.G_FEMALE) {
             searchCat_Gender += "Female/";
         } else {
             searchCat_Gender += "Male/";
         }
         String searchCat_Role = Person.getRoleDesc(p.getPrimaryRole(), false) + "/";
+        String searchCat_RoleGroup = "";
+        String searchCat_CombatSupport = "";
         if (searchCat_Role.startsWith("Admin/")) {
-            searchCat_Role = "Admin/";
+            searchCat_RoleGroup = "Admin/";
         }
-        
+        if (searchCat_Role.endsWith("Tech/") || searchCat_Role.equals("Mechanic/")) {
```

Comment:
Are these role names defined as constants anywhere (like in the Person class)? If so, you should make use of those constants, rather than having magic strings.

Reply:
I believe so. I just went with the same type of hack that was used for `Admin/`, but we could easily replace this with a proper complete enumeration of the matching roles using constants.

Reply:
Ugh, the relevant place to look at is `Person.getRoleDesc()` which takes an int enum and returns a hard-coded string via select case, so each comparison would have to look like `if (searchCat_Role.equals(Person.getRoleDesc(Person.T_MECH_TECH))) { // do; }`. 

Considering this, I think it might be more robust to grab the current person's role int to compare to the int enum, e.g. `if (person.getPrimaryRole() == Person.T_MECH_TECH) { // do; }`.

Reply:
Make it so, #1